etcdserver/api/v3rpc/util.go

```diff
@@ -49,7 +49,7 @@ func togRPCError(err error) error {
 	case etcdserver.ErrTooManyRequests:
 		return rpctypes.ErrTooManyRequests
 
-	case etcdserver.ErrNoLeader:
+	case etcdserver.ErrNoLeader, rpctypes.ErrNoLeader: // ErrNoLeader from leasehttp
```

Comment:
hmm... seem bad. Now API layer needs to know the details of the lease implementation. It now start to understand io related errors. I feel we should still put the error convert back to the HTTP handler itself.